jduong09/thirteen
# AI Difficulties

Resolves #24

## Changes/Additions
- Formed AI logic to generate all possible combination plays for given cards.
- Created hierarchy of hardest to easiest combinations based on probabilities.
- Formed logic to have AI follow previous player with lowest possible value combination that BEATS previous played combination.
- Formed logic 
- If AI has multiple possible plays for a turn combo, AI will choose lowest value combo that beats the previous pile value

## Hot To Test
- Checkout to `issue-24--difficulties` and pull changes with `git pull`
- Navigate to `localhost:3000` in your browser
- Shuffle deck, first turn is either user or AI.
- CPU will try to play valid turn and pass if there is no valid turn.
- Use console logs to verify correct turn and follow along with middle pile as well.
- Game cycle will continue until all cards are played.
  - If AI has no cards in their hand, they will automatically pass. ( NEED LOGIC TO CONFIRM WINNER )

## Notes
  - How to account for when there are multiple combinations of one that is not better than a difficulty greater than it, but playing it would allow it to be played multiple.
  - For example, having 3 pairs compared to 1 three of a kind. If they play the 1 pair, they can follow up with more pairs in the current game cycle, compared to losing their turn lead if a player follows with a three of a kind.
